tests/store/test_sqlalchemy_store.py

```diff
@@ -356,93 +346,78 @@ def test_to_mlflow_entity(self):
 
     def test_delete_run(self):
         run = self._run_factory()
-        self.session.commit()
 
-        run_uuid = run.run_uuid
-        self.store.delete_run(run_uuid)
-        actual = self.session.query(models.SqlRun).filter_by(run_uuid=run_uuid).first()
-        self.assertEqual(actual.lifecycle_stage, entities.LifecycleStage.DELETED)
+        self.store.delete_run(run.info.run_uuid)
+
+        with self.store.ManagedSessionMaker() as session:
+            actual = session.query(models.SqlRun).filter_by(run_uuid=run.info.run_uuid).first()
+            self.assertEqual(actual.lifecycle_stage, entities.LifecycleStage.DELETED)
 
-        deleted_run = self.store.get_run(run_uuid)
-        self.assertEqual(actual.run_uuid, deleted_run.info.run_uuid)
+            deleted_run = self.store.get_run(run.info.run_uuid)
+            self.assertEqual(actual.run_uuid, deleted_run.info.run_uuid)
 
     def test_log_metric(self):
         run = self._run_factory()
 
-        self.session.commit()
-
         tkey = 'blahmetric'
         tval = 100.0
         metric = entities.Metric(tkey, tval, int(time.time()))
         metric2 = entities.Metric(tkey, tval, int(time.time()) + 2)
-        self.store.log_metric(run.run_uuid, metric)
-        self.store.log_metric(run.run_uuid, metric2)
-
-        actual = self.session.query(models.SqlMetric).filter_by(key=tkey, value=tval)
-
-        self.assertIsNotNone(actual)
```

Comment:
This statement, which was the only reference to `actual` is vacuously true, regardless of whether or not the contents of the database match the expected values, because `actual` is an *unexecuted* query of type `sqlalchemy.orm.query.Query`.

Similar logic existed in several test cases. I've removed it because these cases include logic that validates the contents of the store using its public APIs, which seems to be a more appropriate mechanism of verifying that CRUD operations produce expected results.

Reply:
nice catch!